packages/cfd/src/Components/success-dialog/__tests__/success-dialog.spec.tsx

```diff
@@ -0,0 +1,165 @@
+        expect(
+            screen.getByRole('button', {
+                name: 'Cancel',
+            })
+        ).toBeInTheDocument();
+    });
+
+    it('should render submit button when `has_submit` is true and `onSubmit` is provided', () => {
+        render(<SuccessDialog is_open={true} has_submit onSubmit={mockSubmitFn} text_submit='Submit' />);
+
+        screen
+            .getByRole('button', {
+                name: 'Submit',
+            })
+            .click();
+
+        expect(mockSubmitFn).toBeCalled();
+    });
+
+    it('should have medium className for buttons when `is_medium_button` prop is true', () => {
+        render(
+            <SuccessDialog
+                is_open={true}
+                has_cancel
+                onCancel={mockCancelFn}
+                has_submit
+                onSubmit={mockSubmitFn}
+                is_medium_button
+                text_submit='Submit'
+            />
+        );
+
+        const cancelButton = screen.getByRole('button', { name: 'Maybe later' });
+        const submitButton = screen.getByRole('button', { name: 'Submit' });
+
+        expect(cancelButton).toHaveClass('dc-btn__medium');
+        expect(submitButton).toHaveClass('dc-btn__medium');
+    });
+
+    it('should not render submit button when `has_submit` is true but `onSubmit` is not provided', () => {
+        render(<SuccessDialog is_open={true} has_submit text_submit='Submit' />);
+        expect(
+            screen.queryByRole('button', {
+                name: 'Submit',
+            })
+        ).not.toBeInTheDocument();
+    });
+
+    it('should have xlarge icon class based on icon_size prop is xlarge', () => {
+        render(<SuccessDialog is_open={true} icon_size='xlarge' />);
+        expect(screen.getByTestId('dt_cfd_success_modal_icon_wrapper')).toHaveClass(
+            'success-change__icon-area--xlarge'
+        );
+    });
+
+    it('should have large icon class based on icon_size prop is large', () => {
+        render(<SuccessDialog is_open={true} icon_size='large' />);
+        expect(screen.getByTestId('dt_cfd_success_modal_icon_wrapper')).toHaveClass('success-change__icon-area--large');
+    });
+
+    it('should render the first SVG element correctly', () => {
+        render(<SuccessDialog is_open={true} icon={<Icon icon='IcCashierWallet' size={128} data-testid='tttttt' />} />);
+
+        const iconWrapper = screen.getByTestId('dt_cfd_success_modal_icon_wrapper');
+        // eslint-disable-next-line testing-library/no-node-access
+        const svgs = iconWrapper.querySelectorAll('svg');
+
+        // Check that the first SVG has the correct attributes
+        const firstSvg = svgs[0];
+
+        // eslint-disable-next-line testing-library/no-node-access
+        const useElement = firstSvg.querySelector('use');
+        expect(useElement).toHaveAttribute(
+            'xlink:href',
+            '/public/sprites/cashier.bd4595decf98fec30aeb1172bd999274.svg#ic-cashier-wallet'
```

Comment:
I am not sure if expecting a hash value in the test is safe as it might be a dynamic value, how about testing the hash value creation function instead?

Reply:
I mocked the Icon component.